client/socket_client.go

```diff
@@ -61,6 +62,10 @@ func (cli *socketClient) OnStart() error {
 		return err
 	}
 
+	if len(strings.SplitN(cli.addr, "://", 2)) < 2 {
+		return fmt.Error("Missing protocol provided (ex. tcp://)", cli.addr)
```

Comment:
Maybe `Missing protocol (ex. tcp://) in %v`?

Reply:
Love it, will incorporate